### PR TITLE
Fix `gt` column labels

### PR DESCRIPTION
**Pull request by Jeff-Thompson12**
Rewrites `col_for_list_expr()` to behave as expected.

Adds example datasets.

Adds examples to the functions `app_methods()` and `col_for_list_expr()`.

### CODE REVIEW

**Comment by AARON-CLARK** [comment]
@Jeff-Thompson12, I spoke too soon. This branch has a merge conflict with `for_cranny`. Please take a look. Thanks!

**Comment by AARON-CLARK** [comment]
@Jeff-Thompson12, sorry, one more edit request: I'm a big fan of calling out your imports in each of the Rd file. I think I learned that in Hadley's book. So could you add the following `roxygen2` code:
```
#' @importFrom purrr map2
#' @importFrom gt md
#' @importFrom glue glue
#' @importFrom rlang set_names
```

**Comment by Jeff-Thompson12** [comment]
@AARON-CLARK I don't know why the ubuntu tests failed. Do you want to rerun those jobs again before merging?

**Comment by AARON-CLARK** [comment]
@Jeff-Thompson12, Yeah - I'm guessing it's a fluke. A ubuntu test failed on `for_cranny` this AM and it fixed itself with a re-run. I'll kick that off now.

**Comment by AARON-CLARK** [comment]
hmmm @Jeff-Thompson12, checks are still unsuccessful on ubuntu!

**Comment by Jeff-Thompson12** [comment]
Yeah, I don't understand it. It's erroring on an example that outputs a `gt` table object. It's possible that those objects should only be ran interactively, but that doesn't explain why it passed the checks yesterday.

**Comment by Jeff-Thompson12** [comment]
@AARON-CLARK I guess the use of `interactive()` in that example was needed.

File: R/utils_helpers.R [hunk 1]
```diff
@@ -1,15 +1,35 @@
...
+#' @importFrom glue glue
+#' @importFrom rlang set_names
+#' 
+#' @examples 
+#' data(example_dat2, package = "tidyCDISC")
+#' 
+#' labels <- col_for_list_expr(example_dat2$col_names, example_dat2$col_totals)
+#' labels
+#' 
+#' if (interactive()) {
```
Comment (AARON-CLARK) [inline R/utils_helpers.R:23]: Is there a reason the `if(interactive())` is in the middle of the example? Is there any benefit to compiling only half of the example and not the other half? Just wondering if the entire example be encased in the `if(interactive())`?

**Review by AARON-CLARK — COMMENTED** [review]
@Jeff-Thompson12, Looks good! Just one comment:

File: R/utils_helpers.R (hunk 1 above)
Comment (Jeff-Thompson12) [reply R/utils_helpers.R:23]: Yes, while there may not be benefit for CRAN documentation, there is benefit for other documentation like the `pkgdown` website where the code gets compiled. You can see the output of the `labels` that are getting created at least.

Comment (AARON-CLARK) [reply R/utils_helpers.R:23]: Okay, as long as it's somewhat useful. I'll merge this PR and then update my cran comments before re-submitting! Thanks @Jeff-Thompson12

**Pull request merged**